### PR TITLE
Batch cluster state tasks for index opening, closing, and blocks

### LINKED ISSUE #81627: Batch add-block-index-to-close, add-index-block and finalize-index-block tasks
The process of closing an index involves multiple cluster state updates to add and finalise its blocks; there are also other reasons for adding blocks to an index. ILM may close an index to adjust its codec prior to force-merging, and may apply a write block for other reasons. Today the associated cluster state update tasks run at priority `URGENT` and are not batched, so can take a very long time to process and gum up other things.

I encountered a cluster that was processing a few thousand such tasks at a rate of about 20 per minute.

We should batch these tasks together to avoid this causing a problem in future.

Relates https://github.com/elastic/elasticsearch/issues/77466

### PR DESCRIPTION
**Pull request by joegallo**
Closes https://github.com/elastic/elasticsearch/issues/81627

Pretty straightforward -- instead of using `unbatched()` five times, we use a simple batching executor that just loops through the tasks and executes them all.

A few thoughts:

1. `open-indices` uses an `AckedClusterStateUpdateTask` rather than just a `ClusterStateUpdateTask` -- is it still okay to batch those together with the rest of the tasks?
2. For that matter, are there any drawbacks to batching open/close and the index blocks all together? I can't think of any, but I'm curious if anybody else can come up with a reason we shouldn't.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexStateService.java [hunk 1]
```diff
@@ -925,7 +940,7 @@ public ClusterState execute(final ClusterState currentState) {
                     return allocationService.reroute(updatedState, "indices opened [" + indicesAsString + "]");
                 }
             },
-            ClusterStateTaskExecutor.unbatched()
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexStateService.java:928]: Not sure we want to batch this in with the other operations? It seems like this is not something that ILM will do in a hot loop right?
I could see some strange side effects of batching open and close index updates into a single batch in some edge cases.

**Review by original-brownbear — COMMENTED** [review]
Thanks Joe! I'll take a detailed look tomorrow. One small point only after a quick read. I think this might be fine as is but I'll have to think through potential side effects for a bit.

**Comment by joegallo** [comment]
With the callback-accepting simple batching executor I added in ee83da0, we could defer the `allocationService.reroute` in the open and close tasks until the end of their batch.

**Comment by joegallo** [comment]
https://github.com/elastic/elasticsearch/blob/fa95bb38634781a65b151be800bc19094c7c49f0/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ForceMergeAction.java#L185-L190

w.r.t. batching of opens and closes, I'm hesitant to only batch the blocks, because it seems like the blocks are part of the close, and then the close is always followed by an open (at least in the case of ILM). OTOH if the closes and opens are cheaper/faster by an order of magnitude or something like that, then maybe it doesn't matter.

**Comment by joegallo** [comment]
Between #83760, #84259, and #84374, this one is no longer necessary and can be closed.

**Pull request closed without merging**